test/unit/driver/aws-sigv4-driver-remote-connection.test.js

```diff
@@ -204,11 +204,8 @@ describe('AwsSigV4DriverRemoteConnection', () => {
 
   describe('open', () => {
     it('should open the client connection', () => {
-      const connection = new AwsSigV4DriverRemoteConnection(HOST, PORT, OPTS);
-      const open = jest.fn();
-      connection._client = { open };
+      const connection = new AwsSigV4DriverRemoteConnection(HOST, PORT, { ...OPTS, openOnStartup: false }, () => { expect(1).toEqual(1); });
```

Comment:
I don't think this test is running correctly. By the time you get a callback, your test is already done. I'm not familiar with jest, but chai has `chaiAsPromised` which lets you handle async operations such as this.

You should wrap this in a promise and make the promise be resolved in the callback.

Reply:
Good catch, I've just put that `expect(1).toEqual(1)` temporarily and forgot to write it properly.